lib/raxx/server.ex

```diff
@@ -234,19 +236,44 @@ defmodule Raxx.Server do
     normalize_reaction(module.handle_info(other, state), state)
   end
 
-  defp normalize_reaction(response = %Raxx.Response{body: true}, _initial_state) do
+  # Note handle_* different to the handle function,
+  # handle_* is given a server and returns a server, handle is given a server and returns a server state
+  # handle uses the data structures sent from Ace but handle_* uses the data in shape defined my callback
+  def handle_head({module, state}, request = %Raxx.Request{}) do
```

Comment:
@CrowdHailer these need some docs - I copied them from your PR, if you want to take a crack at them to convey best what you have in mind, go for it, otherwise I can whip something up ;)

Reply:
Do you think the args order is the right way round for these functions?

Reply:
I'm not sure, both this and the `Raxx.Server.handle/2` are pretty wonky anyways, but it's fine, because they shouldn't be used too much by the end users.

You could argue for both flipping result tuple order or the argument order, but what we have here looks alright.

Reply:
I added some docs and typespecs